### ISSUE TITLE
Desativação do Webservice SOAP de Preços e Prazos

### ISSUE THREAD

**Issue by jdgrieco** [stavarengo/php-sigep#520]
Chegou a meu conhecimento, através de um gerente comercial de um cliente, que o webservice via SOAP de cálculo de preços e prazo será desativado em 31/08/23.

Alguém está sabendo algo sobre isso?

E os outros endpoints: fechamento de PLP, etc

**Comment by jrmaia** [comment]
Prezado(a) Cliente,

 

     Alinhado ao compromisso de oferecer as melhores soluções aos nossos Clientes, em meados do primeiro semestre de 2022 os Correios disponibilizaram novos ambientes de API’s (Application Programming Interface) para obtenção de informações de preços, prazos e rastreamento. Essas, desenvolvidas em arquitetura (REST), trazem inúmeras vantagens aos usuários, tais como maior segurança, menor tempo de resposta, ambiente escalável, novo e totalmente aderente às principais tecnologias do mercado.

 

     Nesse interim, trabalhamos junto às principais plataformas fornecedoras de tecnologia para e-commerce para que integrassem suas aplicações às novas API’s disponibilizadas pelos Correios, visando cumprir os prazos para desativação da antiga estrutura que fornecia esses serviços.

 

Nesse contexto, informamos que o novo prazo para a desativação das antigas API (ambiente SOAP) e ativação das novas APIs em arquitetura (REST) será dia 17/09/23. Sendo assim, recomendamos que verifique junto à sua área de TI e/ou plataforma integradora/ERP/TMS se já estão integrados aos novos ambientes dos Correios e, caso ainda não estejam, pedimos acionar o Consultor Comercial, Agência de Relacionamento responsável pelo seu contrato, e se for o caso, nossa CAC – Central de Atendimento.

 

Ressaltamos que todos os Clientes detentores de Contrato a partir do pacote Bronze já tem acesso liberado às novas API’s, porém é necessário que criem credenciais no ambiente [Meu Correios](https://meucorreios.correios.com.br/app/index.php) e emitam a senha de consumo de API’s no Correios [Web Service](https://cws.correios.com.br/), condições exigidas para consumo das novas API’s.

 

     Vale destacar ainda que cada plataforma adota procedimentos específicos para uso desse novo modelo de integração, de modo que recomendamos verificar tais procedimentos junto à sua plataforma.

 

     Ressaltamos mais uma vez que contate o seu Consultor Comercial, Agência de Relacionamento dos Correios, e se for o caso, nossa CAC – Central de Atendimento, para esclarecimentos adicionais ou acesse: https://www.correios.com.br/atendimento/developers/correios-developers.

 

Cordialmente,

 

Empresa Brasileira de Correios e Telégrafos.

**Comment by jrmaia** [comment]
Prezado(a) Cliente,


Alinhado ao compromisso de oferecer as melhores soluções aos nossos Clientes, em meados do primeiro semestre de 2022 os Correios disponibilizaram novos ambientes de API’s (Application Programming Interface) para obtenção de informações de preços, prazos e rastreamento. Essas, desenvolvidas em arquitetura (REST), trazem inúmeras vantagens aos usuários, tais como maior segurança, menor tempo de resposta, ambiente escalável, novo e totalmente aderente às principais tecnologias do mercado.


Nesse interim, trabalhamos junto às principais plataformas fornecedoras de tecnologia para e-commerce para que integrassem suas aplicações às novas API’s disponibilizadas pelos Correios, visando cumprir os prazos para desativação da antiga estrutura que fornecia esses serviços.


Nesse contexto, informamos que o novo prazo para a desativação das antigas API (ambiente SOAP) e ativação das novas APIs em arquitetura (REST) será dia 17/09/23. Sendo assim, recomendamos que verifique junto à sua área de TI e/ou plataforma integradora/ERP/TMS se já estão integrados aos novos ambientes dos Correios e, caso ainda não estejam, pedimos acionar o Consultor Comercial, Agência de Relacionamento responsável pelo seu contrato, e se for o caso, nossa CAC – Central de Atendimento.

 
Ressaltamos que todos os Clientes detentores de Contrato a partir do pacote Bronze já tem acesso liberado às novas API’s, porém é necessário que criem credenciais no ambiente Meu Correios e emitam a senha de consumo de API’s no Correios Web Service, condições exigidas para consumo das novas API’s.

 
Vale destacar ainda que cada plataforma adota procedimentos específicos para uso desse novo modelo de integração, de modo que recomendamos verificar tais procedimentos junto à sua plataforma.

 
Ressaltamos mais uma vez que contate o seu Consultor Comercial, Agência de Relacionamento dos Correios, e se for o caso, nossa CAC – Central de Atendimento, para esclarecimentos adicionais ou acesse: https://www.correios.com.br/atendimento/developers/correios-developers.


Cordialmente,


Empresa Brasileira de Correios e Telégrafos.

**Comment by jdgrieco** [comment]
@jrmaia, obrigado, eu só recebi o aviso por telefone por uma funcionaria da empresa, e segundo ela e a gerente da conta o SOAP iria parar em 01/09.

Eu ativei a consulta via REST, e está "caindo" direto, ficando varias horas fora do ar

Alguém ativou? Como está o funcionamento?

**Comment by jdgrieco** [comment]
@jrmaia, não fiz "por fora"

**Comment by rodrigojob** [comment]
Eu sou idiota mesmo!
Achei que aqui estava pronta.
Em 2021 eu comentei sobre isso e perguntei sobre um roadmap para a alteração, me ignoraram e eu não fiz o meu.
Tenho isso numa outra aplicação que eu fiz por fora (igual o @jdgrieco) , mal testei... 
Vamos lá!
Alguém tem um esboço?

**Comment by sauloraizen** [comment]
Essa biblioteca está funcionando depois da atualização da API dos correios?

**Comment by webesistemas** [comment]
pelo que consegui ver aqui esta trazendo o valor mas não esta trazendo o prazo

**Comment by rafilsk0** [comment]
> Eu sou idiota mesmo! Achei que aqui estava pronta. Em 2021 eu comentei sobre isso e perguntei sobre um roadmap para a alteração, me ignoraram e eu não fiz o meu. Tenho isso numa outra aplicação que eu fiz por fora (igual o @jdgrieco) , mal testei... Vamos lá! Alguém tem um esboço?

Amigo, a beleza do OpenSoruce é isso. Se em 2021 você já tinha essa informação, porque não fez nenhuma contribuição? apenas ficou sugando o codigo do cara, certo? Então ta ai o preço que paga por ser folgado. 
Mas fica relaxado, a integração com eles é bem suave.. você só vai ter que abandonar o phpsigep e montar o seu proprio,
Php-Sigep já esta muito obsoleto. Tem muita coisa enfiada dentro que nem funciona mais.

**Comment by rafilsk0** [comment]
E fica de dica aqui, para a galera que esta esperando alguma atualização.
A nova integração do correios esta muito suave e facil. As request/response tudo em json, coisa linda de se fazer.
Creio que vai melhor muito as integrações.

Agora, se vocês vão continuar esperando a atualização de uma library de quase 10 anos para se adequar a algo tão novo e sem ser como uma gambiarra(igual eu falei, com mais codigo obsoleto dentro da biblioteca), me desculpem. Mas dificilmente eu acho que isso vai rolar.

**Comment by prhost** [comment]
https://github.com/devaguia/correios-php

**Comment by claupereirajr** [comment]
Sim. Atualmente somente utilizo a lib para gerar os pdf o restante das funções já separei em uma service própria no laravel

**Comment by rodrigojob** [comment]
Foi erro meu mesmo, tenho outras contribuições neste código, como a Logística Reversa. 
Tive problema para validar um PR e nem sei se foi revisto. Não tive retorno de ninguém sobre o teste de validação com problema de uma variável que nem estava no código (acho que era isso).

O projeto estava bom até setembro, para mim tudo funcionava. Você que é desistente.

> Amigo, a beleza do OpenSoruce é isso. Se em 2021 você já tinha essa informação, porque não fez nenhuma contribuição? apenas ficou sugando o codigo do cara, certo? Então ta ai o preço que paga por ser folgado. Mas fica relaxado, a integração com eles é bem suave.. você só vai ter que abandonar o phpsigep e montar o seu proprio, Php-Sigep já esta muito obsoleto. Tem muita coisa enfiada dentro que nem funciona mais.

**Comment by claupereirajr** [comment]
Vou realizar algumas alterações, pois atualizamos para o PHP 82 e tivemos alguns erros. Mas acho que o caminho que vou seguir é deixar somente para realizar a geração de etiquetas, ar e outros modelos de pdf, até como os Correios pretende remover a PLP pelo que vi na API nova.

**Comment by sauloraizen** [comment]
> Vou realizar algumas alterações, pois atualizamos para o PHP 82 e tivemos alguns erros. Mas acho que o caminho que vou seguir é deixar somente para realizar a geração de etiquetas, ar e outros modelos de pdf, até como os Correios pretende remover a PLP pelo que vi na API nova.

A geração de etiquetas está funcionando nesse projeto?
